#! /bin/sh /usr/share/dpatch/dpatch-run
## 39_nonroot_skips_rezero_unit.dpatch by  <blade@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't run rezero_unit without having root permissions (causing critical error).

--- a~/cdrecord/cdrecord.c	
+++ a/cdrecord/cdrecord.c	
@@ -3939,7 +3939,8 @@
 	scsi_start_stop_unit(scgp, 1, 0, immed);
 	wait_unit_ready(scgp, 120);
 	scgp->silent++;
-	rezero_unit(scgp);	/* Is this needed? Not supported by some drvives */
+	if(geteuid() == 0) // EB: needed? Not allowed for non-root, that is sure.
+      rezero_unit(scgp);	/* Is this needed? Not supported by some drvives */
 	scgp->silent--;
 	test_unit_ready(scgp);
 	scsi_start_stop_unit(scgp, 1, 0, immed);

